jdaviz/configs/default/plugins/export_plot/export_plot.py

```diff
@@ -49,7 +52,15 @@ def save_figure(self, filename=None, filetype=None):
 
         viewer = self.viewer.selected_obj
         if filetype == "png":
-            viewer.figure.save_png(filename)
+            if filename is None:
+                viewer.figure.save_png()
+            else:
+                # support writing without save dialog
+                # https://github.com/bqplot/bqplot/pull/1397
+                def on_img_received(data):
+                    with open(filename, 'bw') as f:

```

Comment:
Maybe we need to wrap filename inside [expanduser](https://docs.python.org/3/library/os.path.html#os.path.expanduser) so `~` works for @rosteen ?